src/main/java/racingcar/Car.java

```diff
@@ -0,0 +1,96 @@
+package racingcar;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import static java.util.Arrays.*;
+import static java.util.stream.Collectors.toList;
+
+public class Car {
+    public static final int START_SCORE = 1;
+    private static final int MAX_CARNAME = 5;
+    private static final int MIN_RANDNUM = 3;
+    private static final String DASH_PRINT_POSITION = "-";
+
+
+    private int position;
+    private final String name;
+
+    public Car(String name, int position) {
+        this.name = name;
+        this.position = position;
+    }
+
+
+    public static List<Car> createCar(String[] carNameList) {
+        validate(carNameList);
+        return stream(carNameList)
+                .map(carName -> new Car(carName, START_SCORE))
+                .collect(toList());
+    }
+
+    public static void validate(String[] carNameList) {
+        Set<String> checker = new HashSet<>();
+
+        for (String carName : carNameList) {
+            validateCarNameLength(carName);
+            checker.add(carName);
+        }
+
+        if (checker.size() != carNameList.length) {
+            throw new IllegalArgumentException("동일한 차량 이름은 불가능합니다.");
+        }
+    }
+
+    private static void validateCarNameLength(String str) {
+        if (str.length() > MAX_CARNAME) {
+            throw new IllegalArgumentException("차량 이름은 5자를 넘어서는 안됩니다.");
+        }
+    }
+
+
+    public static Car updatePosition(Car car, int num) {
+        if (num > MIN_RANDNUM) {
+            car.position++;
+        }
+        return new Car(car.name, car.position);
+    }
+
+    public static Boolean isMaxPosition(Car car, int maxPosition) {
+        return car.getPosition() == maxPosition;
+    }
```

Comment:
메서드 파라미터로 maxPosition이 아닌 position이 들어온다면 지금의 메서드 네이밍과 동일한 동작을 한다고 볼 수 있을까요? 🤔

아래와 같이 변경해서 Car 객체를 객체스럽게(= 자신만의 고유한 상태와 행위를 가지는) 사용해보도록 변경해보시면 좋을 것 같아요. 😃
```suggestion
    public boolean isSamePosition(int position) {
        return this.position == position;
    }
```